### DISCUSSION TITLE
How to get the same target value as in loaderContext.target inside a custom plugin?

### DISCUSSION THREAD

**Discussion by vadymshymko** [webpack/webpack#15912] [Q&A] [answered]
Hello. I have a custom plugin and a custom loader. Inside the plugin's apply method, I want to perform certain actions, but only if the target is the same as inside my custom loader. Getting the target inside the loader is very easy (https://webpack.js.org/api/loaders/#thistarget). But how to get it inside the plugin? Can I rely on the compilation.options.loader.target value?

Code example:

custom loader:
```javascript
module.export = function (source) {
   if (this.target === 'web') {
     // do some job
   }
}
```

custom plugin:
```javascript
compiler.hooks.thisCompilation.tap(
      'MyCustomPluginName',
      (compilation) => {
        const target = compilation.options.loader.target;// is that correct?
        
        compilation.hooks.processAssets.tap(
          'MyCustomPluginName',
          () => {
            if (target === 'web') {//same as in my custom loader
             // do some job
              
```

**Comment by vankop** [comment, marked as answer]
`compilation.options.target`, but probably you need `output.environment` instead of target since target could be e.g. `target="es6"`

**Reply by vadymshymko** [reply]
THNX
